core/src/main/java/org/eclipse/hono/config/FileFormat.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2017 Red Hat Inc and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Red Hat Inc - initial creation
+ */
+
+package org.eclipse.hono.config;
+
+public enum FileFormat {
```

Comment:
Introducing this enum might be a good opportunity to encapsulate the file extension pattern matching in here as well.
WDYT?

Reply:
We could add a method "detect" which takes in the file name and returns an optional FileFormat. Hiding the patterns inside of that implementation.